### PR TITLE
feat(gateway): Do not close `ZeebeClient` in `ResponsiveHealthIndicator`

### LINKED ISSUE #11598: `GatewayResposive` health checks are broken
**Describe the bug**

The main problem is that in [this code](https://github.com/camunda/zeebe/blob/4f8a898f3c63322aa91c5f5768d4e9c1edc6c26c/gateway/src/main/java/io/camunda/zeebe/gateway/impl/probes/health/ResponsiveHealthIndicator.java#L51-L62) we are closing the `ZeebeClient`, so the first check is passed, but the next one will try to use closed `ZeebeClient`. And we will get this response in `/actuator/health`:
```json
{
  "components": {
    ...
    "livenessGatewayResponsive": {
      "details": {
        "derivedFrom": "ResponsiveHealthIndicator",
        "downTime": "PT2M53.443S",
        "lastSeenDelegateHealthStatus": {
          "details": {
            "error": "java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: UNAVAILABLE: Channel shutdown invoked",
  ...
```

**To Reproduce**

1. Run Zeebe Gateway with `gateway-responsive` health checks (enabled by default)
2. Call `/actuator/health` several times with pauses (5 seconds would be enough)
3. Obtain this error

**Expected behavior**

Health checks should not be broken because the `ZeebeClient` is closed.

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: UNAVAILABLE: Channel shutdown invoked
        at java.base/java.util.concurrent.CompletableFuture.reportGet(Unknown Source)
        at java.base/java.util.concurrent.CompletableFuture.get(Unknown Source)
        at io.camunda.zeebe.gateway.impl.probes.health.ResponsiveHealthIndicator.health(ResponsiveHealthIndicator.java:57)
        at io.camunda.zeebe.util.health.DelayedHealthIndicator.checkHealth(DelayedHealthIndicator.java:68)
        at jdk.internal.reflect.GeneratedMethodAccessor62.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:84)
        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: io.grpc.StatusRuntimeException: UNAVAILABLE: Channel shutdown invoked
        at io.grpc.Status.asRuntimeException(Status.java:539)
        at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:487)
        at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:563)
        at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:744)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:723)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
        ... 3 more

```

</p>
</details>

**Environment:**
- OS: Linux 3.10.0-1160.53.1.el7.x86_64
- Zeebe Version: 8.2.0-alpha4 (I suppose that versions from 8.2.0-alpha3 are affected because of [this commit](https://github.com/camunda/zeebe/commit/f93c8dcb96f69c2633d0d91621d2bbe4ecbec107))
- Configuration: 3 brokers, 3 gateways, 5 partitions, Elasticsearch exporter.

### PR DESCRIPTION
**Pull request by aivinog1**
## Description

It calls more than once, so if we close `ZeebeClient` in the next call we got `io.grpc.StatusRuntimeException: UNAVAILABLE: Channel shutdown invoked` exception

## Related issues

closes #11598 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [X] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [X] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by remcowesterhoud** [comment]
@aivinog1 The CI is breaking, could you try rebasing this onto main please?

**Comment by aivinog1** [comment]
> @aivinog1 The CI is breaking, could you try rebasing this onto main please?

Thanks @remcowesterhoud, it is done, can you approve workflows, please? :)

**Comment by remcowesterhoud** [comment]
Ok that didn't fix it, I'll have a deeper look into this.

**Review by remcowesterhoud — APPROVED** [review]
Thanks for the fix @aivinog1! Code all looks good to me.

Because we are still in a discussion how to fix the CI (it's only broken for forks because they can't access our secrets) I have pushed your changes to a branch within our repo. If the CI passes I can merge it 🙂 

https://github.com/camunda/zeebe/pull/11735

File: gateway/src/main/java/io/camunda/zeebe/gateway/impl/probes/health/ResponsiveHealthIndicator.java [hunk 1]
```diff
@@ -48,17 +48,16 @@ HealthZeebeClientProperties getHealthZeebeClientProperties() {
...
-        resultBuilder = Health.unknown();
-      } else {
-        try {
-          zeebeClient.newTopologyRequest().send().get();
-          resultBuilder = Health.up();
-        } catch (final Throwable t) {
-          resultBuilder = Health.down().withException(t);
-        }
+
+    initZeebeClient();
```
Comment (Zelldon) [inline gateway/src/main/java/io/camunda/zeebe/gateway/impl/probes/health/ResponsiveHealthIndicator.java:52]: @remcowesterhoud may I ask why we have the client usage in the gateway? Do you know that? Feels unnecessary dependency and request we sent?

**Comment by remcowesterhoud** [comment]
bors merge

**Pull request merged**